sysdeps/soft-dfp/bid/numdigits.h

```diff
@@ -371,31 +371,37 @@ static inline _Decimal64
 left_justifyd64 (_Decimal64 x)
 {
   int firstdigit = 0, len;
-  char digits[NDIGITS_64+NDIGITS_64-1] = { 0 };
+  char digits[NDIGITS_64 + NDIGITS_64 - 1] = { 0 };
 
   __get_digits_d64 (x, digits, NULL, NULL, NULL, NULL);
   while (digits[firstdigit] == '0')
     firstdigit++;
   len = strlen (digits + firstdigit);
   if (len)
     {
+      int leadingzeroes = -1;
       int exp = getexpd64 (x);
       /* pad the significant digits with enough trailing zeroes */
       if ((exp - firstdigit) < -DECIMAL64_Bias)
 	{
-	  firstdigit = DECIMAL64_Bias + exp;
+
+	  leadingzeroes = DECIMAL64_Bias + exp;	//Here you will left justify the best you can respecting the minimum possible exponent
+
 	  /* If the number overflows the data it will become a NaN1. */
-	  if ((exp - firstdigit) <= -DECIMAL64_Bias && firstdigit != 0)
+	  if (leadingzeroes < 0)
 	    {
-	      union ieee754r_Decimal64 mask = { .di = { 0x00000001,
-                                                        0x7c000000 } };
+	      union ieee754r_Decimal64 mask = {.di = {0x00000001,
+						      0x7c000000}
+	      };
```

Comment:
I would recommend adding a test case (if one is not present yet) to validate this condition for all decimal types.

Reply:
Nice, thanks to call my attention on that. 
I did the math and this case was never going to happen.
What I did is modify the function to check if the number is NaN right in the beginning and return the number. 
https://github.com/libdfp/libdfp/blob/1388913c5b5cd8b1805cea99379ef98334c7135b/sysdeps/soft-dfp/bid/numdigits.h#L344